daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/DSELConverter.java

```diff
@@ -10,25 +10,34 @@
 
 public class DSELConverter {
 
-    private static DSELVisitor visitor = new DSELVisitor();
+    private static final DSELVisitor visitor = new DSELVisitor();
 
     /**
      * Utility method to convert a TQL query to a DSEL query.
      *
      * @param query TQL query as String
-     * @return DSEL ELNode ready to serve for DSEL intepreter
+     * @return DSEL ELNode ready to serve for DSEL interpreter
      */
-    public static ELNode convert(String query) throws TqlException {
-        Expression exp = Tql.parse(query);
-        ELNode raw = exp.accept(visitor);
+    public static ELNode convert(final String query) throws TqlException {
+        final Expression tqlExpression = Tql.parse(query);
+        return convert(tqlExpression);
+    }
 
+    /**
+     * Utility method to convert a TQL query to a DSEL query.
+     *
+     * @param tqlQuery TQL query
+     * @return DSEL ELNode ready to serve for DSEL interpreter
+     */
+    public static ELNode convert(final Expression tqlQuery) throws TqlException {
```

Comment:
TqlElement is not more suitable than Expression ?
```suggestion
    public static ELNode convert(final TqlElement tqlElement) throws TqlException {
```

Reply:
I'm not sure, the old method was casting the result of `Tql#parse` in an `Expression` so I kept it as is. 
Since `Tql#parse` returns an `Expression` I've got the feeling that it's the right level in the class tree but I'll let more experienced people confirm.